include/examples/quadruped_3dof_limb.h

```diff
@@ -0,0 +1,323 @@
+ *
+ * @copyright (c) Copyright 2022 The Trustees of the University of Pennsylvania
+ *
+ */
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include "kodlab_mjbots_sdk/limb_base.h"
+
+namespace quad_example_helpers{
+    
+/**
+ * @brief Creates a 3x3 skew symmetric matrix from a 3-vector
+ *
+ * @param w 3-vector
+ * @return 3x3 skew-symmetric matrix
+ */
+Eigen::Matrix3f Skew(const Eigen::Vector3f &w)
+{
+    Eigen::Matrix3f S;
+    S <<    0, -w(2),  w(1),
+         w(2),     0, -w(0),
+        -w(1),  w(0),     0;
+    return S;
+}
+
+/**
+ * @brief Get the sign of a flag for use in mathematic equations
+ *
+ * @param flag boolean flag
+ * @return 1 if `flag` evaluates to true, -1 otherwise
+ */
+static int GetSign(bool flag) { return (flag ? 1 : -1); }
+
+/**
+     * @brief Solving the equation a0 Sin(th) + a1 Cos(th) = a2
+     *
+     * @param a0 Sin coefficient
+     * @param a1 Cos coefficient
+     * @param a2 Right side coefficient
+     * @return float returns nan if no solution exists
+*/
+float SolveSinCosSum(float a0, float a1, float a2)
+{
+    // a0 Sin(x) + a1 Cos(x) = a2 Solution for x
+    float sqDiff = a0 * a0 + a1 * a1 - a2 * a2;
+    if (sqDiff < 0)
+    { // Check that solution is feasible
+        std::cout << "ERROR NO SOLUTION SinCosSum" << std::endl;
+        return 0.0 / 0.0; // NaN
+    }
+
+    // Mathematica Code:
+    //  ArcTan[a1*a2 + Sqrt[a0^2 + a1^2 - a2^2]*Abs[a0], 
+    //         a0*a2 - (a1*Sqrt[a0^2 + a1^2 - a2^2])/Sign[a0]]
+    float diff = sqrt(sqDiff);
+    return atan2(a0 * a2 - a1 * diff / std::copysignf(1.0, a0), 
+                 a1 * a2 + diff * std::abs(a0));
+}
+
+/**
+ * @brief Wraps an angle -Pi to Pi
+ * @param angle
+ */
+float angle_mod(float angle){
+    return fmod(angle+M_PI, 2*M_PI) - M_PI;
+}
+
+/**
+ * @brief Quick Float Nan
+ * 
+ */
+float qnanf(){
+    return float(0.0/0.0);
+}
+} //namespace example_helpers
+
+/**
+ * @brief A 3-DoF limb based on the common serial linkage in 12DoF quadrupeds
+ * @details Based on the limbs of popular quadrupeds
+ * @warning Needs to be tested, use at your own risk!!!!
+ * @todo    Needs to be tested ^^^
+ * @note    Needs to be tested ^^^
+ */
+class QuadrupedLimb : public kodlab::LimbBase<Eigen::Vector3f> {
+  
+  public:
+    // Limb index is defined as follows
+    typedef uint8_t LimbIndex_t;
+    enum {
+        kFrontLeft = 0,
+        kRearLeft,
+        kFrontRight,
+        kRearRight
+    };
+
+    QuadrupedLimb(
+        const std::vector<std::shared_ptr<kodlab::JointBase>> &joints,
+        const kodlab::LimbConfig &config,
+        LimbIndex_t limb_id = kFrontLeft)
+        : kodlab::LimbBase<Eigen::Vector3f>::LimbBase("", joints, config)
+    {
+        is_front_limb_ = (int(limb_id) % 2) == 0;
+        is_left_limb_ = (int(limb_id) / 2) == 0;
+    }    
+    
+    Eigen::Vector3f ForwardKinematicsImpl(
+        std::vector<float>thetas) override 
+    {      
+        Eigen::Vector3f d0 = config_.actuator_offsets[0];
```

Comment:
The implementation here should be in a `.cpp`. The best way I found to due this with minimal overhead is by default when creating a class to create the source and header.